# Module 4 Lab 2: Textbook Problem 3.1

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("allbacks.csv")
df

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb
5,6,641,367,600,hb
6,7,1228,396,1075,hb
7,8,412,0,250,pb
8,9,953,0,700,pb
9,10,929,0,650,pb


In [3]:
df["cover"].sort_values()

0     hb
1     hb
2     hb
3     hb
4     hb
5     hb
6     hb
7     pb
8     pb
9     pb
10    pb
11    pb
12    pb
13    pb
14    pb
Name: cover, dtype: str

### Median cover type

The median cover type is pb (paperback), the 8th value in the sorted list of 15 cover types.


In [4]:
df["weight"].mean()


np.float64(706.6666666666666)

### Mean weight of all books

The mean weight of all books is 706.67 grams

In [5]:
df.groupby("cover")["weight"].mean()

cover
hb    796.428571
pb    628.125000
Name: weight, dtype: float64

### Mean weight by cover type

The mean weight is 796.43 grams for hardback books and 628.13 grams for paperback books. Hardback books weigh about 168.30 grams more on average. So, we can assume that  paperback books are lighter in this sample.

### By-hand verification

For the verification, all weight values were used as whole numbers.

Mean weight = (800 + 950 + 1050 + 350 + 750 + 600 + 1075 + 250 + 700 + 650 + 975 + 350 + 950 + 425 + 725) / 15

Mean weight = 10,600 / 15 = 706.67 grams.

The by-hand result matches the Python mean weight for all books (average), rounded to 706.67 grams.